cmd/argocd/commands/app.go

```diff
@@ -750,7 +762,38 @@ func NewApplicationUnsetCommand(clientOpts *argocdclient.ClientOptions) *cobra.C
 			errors.CheckError(err)
 
 			updated := false
+			if app.Spec.Source.Kustomize != nil {
+				c.Flags().Visit(func(f *pflag.Flag) {
+					switch f.Name {
+					case "nameprefix":
+						updated = true
+						app.Spec.Source.Kustomize.NamePrefix = ""
+					case "namesuffix":
+						updated = true
+						app.Spec.Source.Kustomize.NameSuffix = ""
+					case "kustomize-image":
+						for _, kustomizeImage := range kustomizeImages {
+							for i, item := range app.Spec.Source.Kustomize.Images {
+								if item.Match(argoappv1.KustomizeImage(kustomizeImage)) {
```

Comment:
Noticed one issue. This check should be implemented as follows (kustomize set expression checks for image name, not visa versa):

```suggestion
								if argoappv1.KustomizeImage(kustomizeImage).Match(item) {
```

Reply:
Fixed.